packages/synapse-client/out/modelGeneric.mustache

```diff
@@ -0,0 +1,174 @@
+import { mapValues } from '../runtime{{importFileExtension}}';
+{{#hasImports}}
+{{#imports}}
+import type { {{{.}}} } from './{{.}}{{importFileExtension}}';
+import {
+    {{.}}FromJSON,
+    {{.}}FromJSONTyped,
+    {{.}}ToJSON,
+} from './{{.}}{{importFileExtension}}';
+{{/imports}}
+
+{{/hasImports}}
+{{#discriminator}}
+import {
+{{#discriminator.mappedModels}}
+     {{modelName}}FromJSONTyped{{^-last}},{{/-last}}
+{{/discriminator.mappedModels}}
+} from './index{{importFileExtension}}';
+
+{{/discriminator}}
+{{>modelGenericInterfaces}}
+
+/**
+ * Check if a given object implements the {{classname}} interface.
+ */
+{{! START CUSTOM TEMPLATING FOR synapse-client }}
+{{! Change the return type from `boolean` to a type predicate so this method can be used for type narrowing }}
+export function instanceOf{{classname}}(value: object): value is {{classname}} {
+{{! END CUSTOM TEMPLATING FOR synapse-client }}
```

Comment:
The return type of this function was originally `boolean` but we can make it a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) so this function can be used as a type guard, e.g.

```
const obj = {
  // ... other properties
  "concreteType":  "org.sagebionetworks.repo.model.FileEntity"
}

if (instanceOfFileEntity(obj)) {
  // in this block, obj is known to be FileEntity and can be treated as such without type casting
}

```

Reply:
type predicate was added in upstream